compat/winansi.c

```diff
@@ -8,6 +8,22 @@
 #include <winreg.h>
 #include "win32.h"
 
+#if defined(_MSC_VER)
```

Comment:
It would be really, really awesome if these changes could be moved into their own commit. I really think that the improved `isatty()` handling deserves its own patch.